direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java

```diff
@@ -229,12 +251,21 @@ private Cluster getCluster(String authority) {
   @VisibleForTesting
   Cluster createCluster(String authority) {
     log.info("Creating cluster for authority {} in accessor {}", authority, this);
-    return Cluster.builder()
-        .addContactPointsWithPorts(
-            Arrays.stream(authority.split(","))
-                .map(CassandraDBAccessor::getAddress)
-                .collect(Collectors.toList()))
-        .build();
+    return configureClusterBuilder(Cluster.builder(), authority).build();
+  }
+
+  @VisibleForTesting
+  Builder configureClusterBuilder(Builder builder, String authority) {
+
+    builder.addContactPointsWithPorts(
+        Arrays.stream(authority.split(","))
+            .map(CassandraDBAccessor::getAddress)
+            .collect(Collectors.toList()));
+    if (username != null) {
+      Preconditions.checkArgument(password != null, "Password must be specified.");
```

Comment:
Is it possible that password can be empty? Can we provide empty password in the configuration, or will it be `null` here?

Reply:
Make sense, changed.